indent/handlebars.vim

```diff
@@ -79,16 +81,55 @@ function! GetHandlebarsIndent(...)
   " all indent rules only apply if the block opening/closing
   " tag is on a separate line
 
-  " indent after block {{#block
+  " check for a hanging attribute
+  let lastLnumCol = col([lnum, '$']) - 1
+  if synIDattr(synID(lnum, lastLnumCol, 1), "name") =~ '^mustache'
+        \ && prevLine !~# '}}\s*$'
+    let hangingAttributePattern = '{{[#^]\=\%(\k\|[/-]\)\+\s\+\zs\k\+='
+    let standaloneComponentPattern = '^\s*{{\%(\k\|[/-]\)\+\s*$'
+
+    if prevLine =~ hangingAttributePattern
+      " {{component attribute=value
+      "             other=value}}
+      let [line, col] = searchpos(hangingAttributePattern, 'Wbn', lnum)
+      if line == lnum
+        return col - 1
+      endif
+    elseif prevLine =~ standaloneComponentPattern
+      " {{component
+      "   attribute=value}}
+      return indent(lnum) + sw
+    endif
+  endif
+
+  " check for a closing }}, indent according to the opening one
+  let saved_pos = getpos('.')
+  if prevLine =~# '}}$' && prevLine !~# '^\s*{{' &&
+        \ currentLine !~# s:componentClosingPattern &&
+        \ search('}}$', 'Wb')
+    let [line, col] = searchpairpos('{{', '', '}}', 'Wb')
+    if line > 0
+      if strpart(getline(line), col - 1, 3) =~ '{{[#^]'
+        " then it's a block component, indent a shiftwidth more
+        return indent(line) + sw
+      else
+        return indent(line)
+      endif
```

Comment:
```suggestion
      if strpart(getline(line), col - 1, 3) !=~ '{{[#^]'
        return indent(line)
      endif
```

We could almost do this, see what I mean? the `indent(line) + sw` for "normal" indents is happening later anyway, so it would be nice to keep that there.